### PR TITLE
Add device trust "token move check" protos

### PR DESCRIPTION
**Pull request by codingllama**
Add the DeviceConfirmationToken proto, the ConfirmDeviceWebAuthentication RPC, and do the necessary proto changes to implement the confirm step.

A DeviceConfirmationToken is acquired as a result of a successful on-behalf-of authentication attempt (via AuthenticateDevice) and exchanged via a (to be created) Web API endpoint, which hits the ConfirmDeviceWebAuthentication RPC.

See [RFD 0009e][1].

https://github.com/gravitational/teleport.e/issues/3236

[1]: https://github.com/gravitational/teleport.e/blob/master/rfd/0009e-device-trust-web-support.md#the-token-move-check

### CODE REVIEW

File: proto/teleport/lib/teleterm/v1/service.proto [hunk 1]
```diff
@@ -655,7 +656,12 @@ message AuthenticateWebDeviceRequest {
   // Device web token to be spent in exchange for the device authentication
   // attempt.
   teleport.devicetrust.v1.DeviceWebToken device_web_token = 1;
+  string cluster_uri = 2;
```
Comment (codingllama) [inline proto/teleport/lib/teleterm/v1/service.proto:659]: This was just missing from the initial iteration, but it only dawned on me when I got to writing the RPC. I don't want to do a breaking change, so tag number 2 it is.

File: proto/teleport/lib/teleterm/v1/service.proto
Comment (codingllama) [inline proto/teleport/lib/teleterm/v1/service.proto:1]: FYI @ravicious for a couple of Connect proto tweaks.

File: proto/teleport/lib/teleterm/v1/service.proto (hunk 1 above)
Comment (gzdunek) [reply proto/teleport/lib/teleterm/v1/service.proto:659]: Is it possible to have the URI of a leaf cluster here? I believe it's always a root cluster, so the field should be named `root_cluster_uri`.

**Review by gzdunek — APPROVED** [review]

File: proto/teleport/lib/teleterm/v1/service.proto (hunk 1 above)
Comment (codingllama) [reply proto/teleport/lib/teleterm/v1/service.proto:659]: Good call, done.

**Review by avatus — APPROVED** [review]

**Pull request merged**